src/api/app/controllers/public_controller.rb

```diff
@@ -95,6 +95,14 @@ def project_index
     pass_to_backend path
   end
 
+  # GET /public/source
+  def projects
+    @projects = Rails.cache.fetch(['projectlist', Project.maximum(:updated_at), User.current.forbidden_project_ids]) do
+      @projects = Project.pluck(:name).sort
+    end
+    render 'source/projects'
+  end
```

Comment:
I would prefer to directly route to source#projects instead. If that doesn't work because of authentification or before filter, I would at least move the code that fetches the project names to one central place. Probably the model.
Otherwise both actions easily get out of sync.